src/chapter_1.md

```diff
@@ -1 +1,243 @@
+└── test-dynamic-plugins/
+```
+
+So let's create our plugin using:
+
+```bash
+cargo init --lib my-plugin
+```
+
+Now, go to the Cargo.toml at the root and you should see this new library:
+
+```toml
+members = [
+    "alumet",
+    "my-plugin",
+]
+```
+
+Now, you can fulfil the TOML of the newly created library with data you want. For example:
+
+```toml
+[package]
+name = "my-plugin"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+alumet = { path = "../alumet" }
+```
+
+### Implement MyPlugin
+
+Let's go to the newly created folder containing the new library. We will use the lib.rs file.
+
+To define our plugin, we need to create a Rust structure: **MyPlugin**. This structure will contain all necessary for the plugin to work.
+Let's take an easy structure having 2 fields: config and metrics. Config will contain the configuration of the plugin and metrics which
+will contain all related metrics.
+
+```rust,ignore 
+{{#rustdoc_include ../code/plugin_example.rs:MyPlugin_Struct}}
+```
+
+Let's define the Metrics structure:
+
+```rust,ignore
+{{#rustdoc_include ../code/plugin_example.rs:Metrics}}
+```
+
+For now, the Metrics structure only contains field: *a_metric*. This is a TypedMetricId and its type is an *u64*
+
+### Implement Config
+
+As you can see, MyPlugin contains a Config value. This Config is a structure where you can define value of configuration for the plugin.
+Let's define it:
+
+```rust,ignore
+{{#rustdoc_include ../code/plugin_example.rs:Config}}
+```
+
+The poll_interval will be the time between two measurements. Feel free to add new element in the configuration if needed.
+
+For Alumet a Configuration structure needs to implement the `Default` trait, which define the default value if not modified by the user.
+Let's do it:
+
+```rust,ignore
+{{#rustdoc_include ../code/plugin_example.rs:impl_default_config}}
+```
+
+The default value of poll_interval is a duration of 1 second.
+
+### Implement AlumetPlugin
+
+First, let's create a MyPluginSource struct:
+
+```rust,ignore
+{{#rustdoc_include ../code/plugin_example.rs:MyPluginSource}}
+```
+
+We have a structure: **MyPlugin** let's implement the `AlumetPlugin` trait, this will transform our structure in an Alumet plugin
+defining some functions:
+
+- name()
+- version()
+- init()
+- default_config()
+- start()
+- stop()
+
+Let's define these for our plugin:
+
+```rust,ignore
+{{#rustdoc_include ../code/plugin_example.rs:implAlumetPlugin}}
+```
+
+Let's focus on the start function.
+We want to create a new metric to match with the Metrics structure's field. In this structure, we have one field: *a_metric*.
+First, we create a unit associated with the metric:
+
+```rust,ignore
+{{#rustdoc_include ../code/plugin_example.rs:createPrefixedUnit}}
+```
+
+Then, we use the create_metric() function of the alumet::plugin::AlumetStart. We specify the kind of value (u64), the name
+of the metric, its unit (created above) and the last argument is the description:
```

Comment:
Now that you no longer create the `PrefixedUnit` explicitely (which is good), you need to update the description ^^